<launch>

    <!-- Broadcasting tf from camera to base link for viso2 -->
    <node name="camera1_tf_broadcaster" pkg="phobos_rover_vision" type="tf_broadcaster">
        <param name="parent_link" value="base_link"/>
        <param name="child_link" value="camera1"/>
        <param name="pose_x" value =" 0.265"/>
        <param name="pose_y" value =" 0.035"/>
        <param name="pose_z" value ="-0.070"/>
        <param name="orient_x" value ="-0.5"/>
        <param name="orient_y" value =" 0.5"/>
        <param name="orient_z" value ="-0.5"/>
        <param name="orient_w" value =" 0.5"/>
    </node>

  <!-- Run the stereo image proc -->
    <node name="stereo_image_proc" pkg="stereo_image_proc" type="stereo_image_proc" output="screen" ns="/rover">
        <param name="~approximate_sync" value="True"/>

        <remap from="/rover/left/image_raw" to="/rover/camera1/image_raw" />
        <remap from="/rover/left/camera_info" to="/rover/camera1/camera_info" />

        <remap from="/rover/right/image_raw" to="/rover/camera2/image_raw" />
        <remap from="/rover/right/camera_info" to="/rover/camera2/camera_info"/>
    </node>

    <!-- Viso2 -->
    <node pkg="viso2_ros" type="stereo_odometer" name="stereo_odometer">
        <remap from="image" to="image_raw" />

        <remap from="/stereo/left/image_raw" to="/rover/camera1/image_raw" />
        <remap from="/stereo/left/camera_info" to="/rover/camera1/camera_info" />

        <remap from="/stereo/right/image_raw" to="/rover/camera2/image_raw" />
        <remap from="/stereo/right/camera_info" to="/rover/camera2/camera_info"/>

        <param name="base_link_frame_id" value="/base_link"/>
        <param name="sensor_frame_id" value="/camera1"/>
    </node>

</launch>
